### PR TITLE
move node-xmpp to a sub package

### PR DESCRIPTION
**Pull request by sonnyp**
Alright after playing quite a lot I decided to go with an incremental approach for ["xmpp2"](https://github.com/node-xmpp/xmpp2/issues/1).
The first step is to switch to a monorepo. 

This is the first move in that direction. It moves the code to packages/node-xmpp

Once this is r+ and merged I'll "import" the rest node-xmpp-client/node-xmpp-component/... into packages/{}

We'll use lerna to manage that monorepo.
https://github.com/lerna/lerna

@astro @lloydwatkin  review?

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,9 +0,0 @@
-examples/
-test/
-
-.editorconfig
-.gitingore
-.npmignore
-.travis.yml
-
-CONTRIBUTING.md
```
Comment (sonnyp) [inline .npmignore]: `/` is private and won't be published

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,6 @@ node_js:
   - '0.12'
   - '4'
   - '5'
-  - 'stable'
+  - '6'
```
Comment (sonnyp) [inline .travis.yml]: I wished travis did that on its own...

File: .travis.yml [hunk 3]
```diff
@@ -7,6 +7,6 @@ node_js:
...
 
-before_script: npm install -g mocha standard
+before_script: npm run bootstrap
```
Comment (sonnyp) [inline .travis.yml]: https://github.com/lerna/lerna#bootstrap

File: .travis.yml [hunk 4]
```diff
@@ -3,10 +3,9 @@ sudo: false
 language: node_js
 
 node_js:
-  - '0.10'
```
Comment (sonnyp) [inline .travis.yml]: lerna depends on execSync and who cares about Node.js 0.10 anymore anyway?

File: packages/node-xmpp/test/tls-test.js [hunk 5]
```diff
@@ -118,7 +118,9 @@ describe('TLS', function () {
         done(new Error('should not allow any authentication'))
       })
 
-      cl.once('error', function () {
+      cl.once('error', function (err) {
+        // assert(err instanceof Error) FIXME should be an instance of Error
+        assert.equal(err.toString(), 'XMPP authentication failure')
         done()
       })
     })
```
Comment (sonnyp) [inline packages/node-xmpp/test/tls-test.js]: TODO later those tests should be moved to a private sub package "node-xmpp-tests"

File: .npmignore (hunk 1 above)
Comment (lloydwatkin) [inline .npmignore]: If you have a parent `package.json` you can also specify `private: true` to prevent it getting published by accident

Comment (sonnyp) [inline .npmignore]: it is already specified in `/package.json`

**Comment by sonnyp** [comment]
@astro @lloydwatkin you ok with this?

File: .travis.yml (hunk 4 above)
Comment (sonnyp) [inline .travis.yml]: Node.js 0.10 support was removed from server master

**Pull request merged**